autogpt_platform/backend/backend/server/v2/store/db.py

```diff
@@ -989,170 +1123,144 @@ async def get_admin_submissions(
+                where_dict["OR"].append({"owningUserId": {"in": user_ids}})
 
         # Calculate pagination
         skip = (page - 1) * page_size
 
-        # Query submissions from database - use the StoreSubmission view
-        submissions = await prisma.models.StoreSubmission.prisma().find_many(
-            where=where,
-            skip=skip,
-            take=page_size,
-            order=[{"date_submitted": "desc"}],
-        )
-
-        # Get total count for pagination
-        total = await prisma.models.StoreSubmission.prisma().count(where=where)
-        total_pages = (total + page_size - 1) // page_size
-
-        # Convert to response models - using all fields for admins
-        submission_models = []
-        for sub in submissions:
-            submission_model = backend.server.v2.store.model.StoreSubmission(
-                agent_id=sub.agent_id,
-                agent_version=sub.agent_version,
-                name=sub.name,
-                sub_heading=sub.sub_heading,
-                slug=sub.slug,
-                description=sub.description,
-                image_urls=sub.image_urls or [],
-                date_submitted=sub.date_submitted or datetime.now(tz=timezone.utc),
-                status=sub.status,
-                runs=sub.runs or 0,
-                rating=sub.rating or 0.0,
-                store_listing_version_id=sub.store_listing_version_id,
-                reviewer_id=sub.reviewer_id,
-                review_comments=sub.review_comments,
-                internal_comments=sub.internal_comments,
-                reviewed_at=sub.reviewed_at,
-                changes_summary=sub.changes_summary,
-            )
-            submission_models.append(submission_model)
-
-        logger.debug(f"Found {len(submission_models)} submissions for admin")
-        return backend.server.v2.store.model.StoreSubmissionsResponse(
-            submissions=submission_models,
-            pagination=backend.server.v2.store.model.Pagination(
-                current_page=page,
-                total_items=total,
-                total_pages=total_pages,
-                page_size=page_size,
-            ),
-        )
-    except Exception as e:
-        logger.error(f"Error fetching admin store submissions: {e}")
-        # Return empty response rather than exposing internal errors
-        return backend.server.v2.store.model.StoreSubmissionsResponse(
-            submissions=[],
-            pagination=backend.server.v2.store.model.Pagination(
-                current_page=page,
-                total_items=0,
-                total_pages=0,
-                page_size=page_size,
+        # Create proper Prisma types for the query
+        where = prisma.types.StoreListingWhereInput(**where_dict)
+        include = prisma.types.StoreListingInclude(
+            Versions=prisma.types.FindManyStoreListingVersionArgsFromStoreListing(
+                order_by=prisma.types._StoreListingVersion_version_OrderByInput(
```

Comment:
nit: we could've just use dict, to make it less intimidating, but this is totally fine :D